### PR TITLE
read root cert from PROV_CERT to support VM

### PR DESCRIPTION
**Pull request by irisdingbj**
This flag is currently used only for VMs.  You can define the provided-cert via `PROV_CERT=/etc/certs` in sidecar.env to plugin the root cert , cert chain and private key for VM. 
Otherwise you must have root cert specified in `/var/run/secrets/istio/root-cert.pem`

### CODE REVIEW

File: tools/packaging/common/envoy_bootstrap_v2.json
Comment (howardjohn) [inline tools/packaging/common/envoy_bootstrap_v2.json]: Isn't OutputKeyCertToDir meant for debugging/alternative usage of the certs like prometheus?? Why are we using it ourselves? 

@lei-tang @costinm

Comment (irisdingbj) [reply tools/packaging/common/envoy_bootstrap_v2.json]: we used it internally as well to see whether we have provided the cert by ourself:  
`sa := istio_agent.NewSDSAgent(proxyConfig.DiscoveryAddress, controlPlaneAuthEnabled, pilotCertProvider, jwtPath, outputKeyCertToDir)
`
If we have such dir we need expect the root-cert exists in that directory. Now it will expects root cert exists both in outputKeyCertToDir and /var/run/secrets/istio

Comment (lei-tang) [reply tools/packaging/common/envoy_bootstrap_v2.json]: OutputKeyCertToDir was originally for the location to which Istio Agent writes a key and certificate for sharing with a service. 
It seems that OutputKeyCertToDir is later extended for VM certificates. Under the PR description section, can you provide a description of the workflows related to this PR? It will help us understand the PR.

**Comment by howardjohn** [comment]
My concern is this will impact non-VM cases

**Comment by lei-tang** [comment]
The impact of this PR is that when {{ if .output_key_cert_to_dir }} is true, the validation CA file will be from .output_key_cert_to_dir, which may break non-VM cases.

**Comment by irisdingbj** [comment]
@lei-tang @howardjohn Thanks for the review. Master branch relies on https://github.com/istio/istio/pull/21658 so I need to work after that get merged and will use `PROP_CERT` instead

**Comment by hzxuzhonghu** [comment]
@irisdingbj This test is being fixed

**Comment by irisdingbj** [comment]
> @irisdingbj This test is being fixed

Thanks for this info! can you point me the fix link?

**Comment by irisdingbj** [comment]
@lei-tang @howardjohn PTAL, Thanks.

File: tools/packaging/common/envoy_bootstrap_v2.json
Comment (lei-tang) [inline tools/packaging/common/envoy_bootstrap_v2.json]: Is .provided_cert only used for the VM case?  Can you add a description of the intended use case or workflow for .provded_cert to the PR description/code?

Comment (irisdingbj) [reply tools/packaging/common/envoy_bootstrap_v2.json]: added description in this PR to describe the flag usage. Please take another look. Thanks.

**Review by lei-tang — APPROVED** [review]

File: tools/packaging/common/envoy_bootstrap_v2.json
Comment (lei-tang) [reply tools/packaging/common/envoy_bootstrap_v2.json]: Thanks! Approved.

Comment (howardjohn) [inline tools/packaging/common/envoy_bootstrap_v2.json]: nit: ProvCert is is `provisioned_cert` not `provided_cert` I believe

File: tools/packaging/common/envoy_bootstrap_v2.json
Comment (howardjohn) [inline tools/packaging/common/envoy_bootstrap_v2.json]: nit: don't make a variable here. Can you just do `"filename": {{(printf "%s%s" .provided_cert "/root-cert.pem") }}`

File: tools/packaging/common/envoy_bootstrap_v2.json
Comment (irisdingbj) [reply tools/packaging/common/envoy_bootstrap_v2.json]: modified to `provisioned_cert`

File: tools/packaging/common/envoy_bootstrap_v2.json
Comment (irisdingbj) [reply tools/packaging/common/envoy_bootstrap_v2.json]: fixed

**Comment by irisdingbj** [comment]
@howardjohn Can you approve it if no problems?

File: tools/packaging/common/envoy_bootstrap_v2.json
Comment (howardjohn) [inline tools/packaging/common/envoy_bootstrap_v2.json]: nit: can you change 
```
pkg/test/framework/components/echo/docker/workload.go
154:                    "PROV_CERT="+istio_agent.CitadelCACertPath,
```

To use a different path? Then we can appropriately test that you can use something other than `./var/run/secrets/istio`

Alternatively, do we even **want** to support other locations?

Comment (irisdingbj) [reply tools/packaging/common/envoy_bootstrap_v2.json]: with this change, we can let user define any location they like.

**Comment by irisdingbj** [comment]
@howardjohn PTAL, Thanks!

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
